client/debug/debug.go

```diff
@@ -75,38 +77,38 @@ func PubkeyCmd() *cobra.Command {
 }
 
 func AddrCmd() *cobra.Command {
-	return &cobra.Command{
+	cmd := &cobra.Command{
 		Use:   "addr [address]",
 		Short: "Convert an address between hex and bech32",
 		Long:  "Convert an address between hex encoding and bech32.",
 		Example: fmt.Sprintf(
-			`$ %s debug addr ethm10jmp6sgh4cc6zt3e8gw05wavvejgr5pw2unfju
-$ %s debug addr 0xA588C66983a81e800Db4dF74564F09f91c026351`, version.AppName, version.AppName),
+			`$ %s debug addr evmos1qqqqhe5pnaq5qq39wqkn957aydnrm45sdn8583
+$ %s debug addr 0x00000Be6819f41400225702D32d3dd23663Dd690 --prefix evmos`, version.AppName, version.AppName),
 		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			addrString := args[0]
-			cfg := sdk.GetConfig()
-
-			var addr []byte
 			switch {
 			case common.IsHexAddress(addrString):
-				addr = common.HexToAddress(addrString).Bytes()
-			case strings.HasPrefix(addrString, cfg.GetBech32ValidatorAddrPrefix()):
-				addr, _ = sdk.ValAddressFromBech32(addrString)
-			case strings.HasPrefix(addrString, cfg.GetBech32AccountAddrPrefix()):
-				addr, _ = sdk.AccAddressFromBech32(addrString)
+				hexAddr := common.HexToAddress(addrString).Bytes()
+				prefix, _ := cmd.Flags().GetString(flagPrefix)
+				bech32Address, _ := sdk.Bech32ifyAddressBytes(prefix, hexAddr)
+
+				cmd.Println("Address bytes:", hexAddr)
+				cmd.Printf("Bech32 %s\n", bech32Address)
 			default:
-				return fmt.Errorf("expected a valid hex or bech32 address (acc prefix %s), got '%s'", cfg.GetBech32AccountAddrPrefix(), addrString)
-			}
+				prefix := strings.SplitN(addrString, "1", 2)[0]
```

Comment:
Can we keep the current way of printing every address by default?  If the prefix flag exists we can just display that one, or add to the list

Reply:
I love the idea of being able to use the pubkey. 
I agree with @ramacarlucho, I also prefer that we have all the formats displayed by default and if the user set the flag, just display the one requested